site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -163,5 +200,124 @@ Site findRootSite() throws ResponseStatusException {
                 .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, Services.NO_ROOT_SITE.message()));
     }
 
+    private SiteDTO convertSite(Site site) {
+        var siteDTO = new SiteDTO(site.getSiteId(), site.getParentSiteId());
+        if (Objects.nonNull(site.getAddress())) {
+            siteDTO.setAddress(convertAddress(site.getAddress()));
+        }
+        siteDTO.setSiteType(site.getSiteType());
+        siteDTO.setAlias(site.getAlias());
+        siteDTO.setName(site.getName());
+        siteDTO.setDescription(site.getDescription());
+        siteDTO.setStatus(site.getStatus());
+        siteDTO.setLastUpdated(site.getLastUpdated());
+
+        return siteDTO;
+    }
+
+    private SiteAddressDTO convertAddress(SiteAddress siteAddress) {
+        return new SiteAddressDTO(siteAddress.getAddress1(), siteAddress.getAddress2(), siteAddress.getAddress3(),
+            siteAddress.getAddress4(), siteAddress.getAddress5(), siteAddress.getCity(), siteAddress.getCountry(),
+            siteAddress.getPostcode());
+    }
+
+    @Override
+    public boolean filterUserSites(List<SiteDTO> sitesReturnObject,
```

Comment:
I think this method was like this before you moved it Darren so not your changes, but does it break the SRP by doing 2 things (1) check authz returning true/false (2) filtering sites ?  Also it seems to operate on the sitesReturnObject argument - side effect?

Reply:
Is it worth disentangling them?  Or -halfway house - returning a Pair to make clear that there are two things going on?   Not a blocker - have approved.